apps/dashboard/lib/trpc/routers/deploy/project/list.ts

```diff
@@ -1,30 +1,68 @@
-import { db } from "@/lib/db";
+import type { Deployment } from "@/lib/collections/deploy/deployments";
+import type { Project } from "@/lib/collections/deploy/projects";
+import { db, sql } from "@/lib/db";
 import { ratelimit, requireUser, requireWorkspace, t, withRatelimit } from "@/lib/trpc/trpc";
-import { TRPCError } from "@trpc/server";
+import { deployments, domains, projects } from "@unkey/db/src/schema";
+
+type ProjectRow = {
+  id: string;
+  name: string;
+  slug: string;
+  updated_at: number | null;
+  git_repository_url: string | null;
+  live_deployment_id: string | null;
+  git_commit_message: string | null;
+  git_branch: string | null;
+  git_commit_author_name: string | null;
+  git_commit_timestamp: number | null;
+  runtime_config: Deployment["runtimeConfig"] | null;
+  domain: string | null;
+};
 
 export const listProjects = t.procedure
   .use(requireUser)
   .use(requireWorkspace)
   .use(withRatelimit(ratelimit.read))
   .query(async ({ ctx }) => {
-    return await db.query.projects
-      .findMany({
-        where: (table, { eq }) => eq(table.workspaceId, ctx.workspace.id),
-        columns: {
-          id: true,
-          name: true,
-          slug: true,
-          updatedAt: true,
-          gitRepositoryUrl: true,
-          liveDeploymentId: true,
-        },
-      })
-      .catch((error) => {
-        console.error("Error querying projects:", error);
-        throw new TRPCError({
-          code: "INTERNAL_SERVER_ERROR",
-          message:
-            "Failed to retrieve projects due to an error. If this issue persists, please contact support.",
-        });
-      });
+    const result = await db.execute(sql`
```

Comment:
what's the benefit of doing a raw sql, rather than a nice-to-read drizzle query?